### ISSUE TITLE
`Layout/RedundantLineBreak` should have option to selectively apply

### ISSUE THREAD

**Issue by duongdominhchau** [rubocop/rubocop#14968]
## Is your feature request related to a problem? Please describe.

Look at this code block:

```rb
let(:a) { create(:something, a:, b:, c:) }

before
  do_something
end
```

Here we have a `let` call and a `before` call. Under the view of current `Layout/RedundantLineBreak` cop, both are the same and are collapsed into single-line blocks.

Collapsing the `let` block is great, as normally we don't use multi-line block for `let` unless absolutely needed. Meanwhile, the `before` block should be multi-line because we may add more actions there, and `do`/`end` is also semantically better at expressing that we are about to do something (action, not expression).

## Describe the solution you'd like

An option (or two options for allow and deny) to match against the receiver of the block to see if `Layout/RedundantLineBreak` should inspect the block to collapse. Can be named `IgnoreInspectPatterns`, like in `Layout/LineLength` we have `AllowedPatterns`.

This way, I can configure:

```yaml
Layout/RedundantLineBreak:
  Enabled: true
  InspectBlocks: true
  IgnoreInspectPatterns:
    - before
    - after
    - around
```

or if we use allowlist:

```yaml
Layout/RedundantLineBreak:
  Enabled: true
  InspectBlocks: true
  AllowedInspectPatterns:
    - let
    - let!
```

Personally I prefer having both, it gives finer control over the `Layout/RedundantLineBreak` cop.

## Describe alternatives you've considered

Have a rule to enforce that for certain block receivers, the block will always be written using `do`/`end`, like this:

```yaml
Layout/EnforceMultilineBlock: # or should this be `Style/EnforceMultilineBlock`?
  Enabled: true
  Receivers:
    - before
    - after
    - around
    - each
    # etc.
```

Then somehow make it run after `Layout/RedundantLineBreak` so the collapsed block is re-expand into `do`/`end` block.
